test/remote/gateways/remote_securion_pay_test.rb

```diff
@@ -104,89 +164,103 @@ def test_failed_authorize
   def test_failed_capture
     response = @gateway.capture(@amount, 'invalid_authorization_token')
     assert_failure response
-    assert_match %r{Requested Charge does not exist}, response.message
+    assert_match %r{Charge 'invalid_authorization_token' does not exist}, response.message
   end
 
   def test_successful_full_refund
-    purchase = @gateway.purchase(@amount, @credit_card, @options)
-    assert_success purchase
-    assert purchase.authorization
+    [@gateway, @gateway_shift4_v2].each do |gateway|
+      purchase = gateway.purchase(@amount, @credit_card, @options)
+      assert_success purchase
+      assert purchase.authorization
 
-    refund = @gateway.refund(@amount, purchase.authorization)
-    assert_success refund
+      refund = gateway.refund(@amount, purchase.authorization)
+      assert_success refund
 
-    assert refund.params['refunded']
-    assert_equal 0, refund.params['amount']
-    assert_equal 1, refund.params['refunds'].size
-    assert_equal @amount, refund.params['refunds'].map { |r| r['amount'] }.sum
+      assert refund.params['refunded']
+      assert_equal 0, refund.params['amount']
+      assert_equal 1, refund.params['refunds'].size
+      assert_equal @amount, refund.params['refunds'].map { |r| r['amount'] }.sum
 
-    assert refund.authorization
+      assert refund.authorization
+    end
   end
 
   def test_successful_partially_refund
-    purchase = @gateway.purchase(@amount, @credit_card, @options)
-    assert_success purchase
-    assert purchase.authorization
-
-    first_refund = @gateway.refund(@refund_amount, purchase.authorization)
-    assert_success first_refund
-
-    second_refund = @gateway.refund(@refund_amount, purchase.authorization)
-    assert_success second_refund
-    assert second_refund.params['refunded']
-    assert_equal @amount - 2 * @refund_amount, second_refund.params['amount']
-    assert_equal 2, second_refund.params['refunds'].size
-    assert_equal 2 * @refund_amount, second_refund.params['refunds'].map { |r| r['amount'] }.sum
-    assert second_refund.authorization
+    [@gateway, @gateway_shift4_v2].each do |gateway|
```

Comment:
💬 Comment:

I think that including both modes for testing is great, but building the gateway collection every time adds a little bit of noise `[@gateway, @gateway_shift4_v2]`, perhaps adding  "@gateways" instance variable for this would be better.

Reply:
you're right, we'll update it.